Kudu.Core/LinuxConsumption/StorageClient.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Diagnostics.Tracing;
+using System.Threading.Tasks;
+using Kudu.Core.Tracing;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.File;
+
+namespace Kudu.Core.LinuxConsumption
+{
+    public class StorageClient : IStorageClient
+    {
+        private readonly ISystemEnvironment _environment;
+
+        public StorageClient(ISystemEnvironment environment)
+        {
+            _environment = environment;
+        }
+
+        public async Task CreateFileShare(string siteName, string connectionString, string fileShareName)
+        {
+            try
+            {
+                var storageAccount = CloudStorageAccount.Parse(connectionString);
+                var fileClient = storageAccount.CreateCloudFileClient();
+
+                // Get a reference to the file share we created previously.
+                CloudFileShare share = fileClient.GetShareReference(fileShareName);
+
+                KuduEventGenerator.Log(_environment).LogMessage(EventLevel.Informational, siteName,
+                    $"Creating Kudu mount file share {fileShareName}", string.Empty);
+
+                await share.CreateIfNotExistsAsync(new FileRequestOptions(), new OperationContext());
```

Comment:
Would this cause the container to recycle ?

Reply:
No. We grant additional capabilities to the container when running on Azure to be able create and mount fileshares without needing to recycle.